lib/active_shipping/carriers/ups.rb

```diff
@@ -450,15 +450,26 @@ def build_shipment_request(origin, destination, packages, options={})
             end
           end
 
-          # I don't know all of the options that UPS supports for labels
-          # so I'm going with something very simple for now.
+          # Supported image_formats:
+          # GIF, EPL, ZPL, STARPL and SPL
+          label_format = options[:label_format].nil? ? 'GIF' : options[:label_format].upcase
+          label_size = options[:label_size].nil? ? [4, 6] : options[:label_size]
+
           xml.LabelSpecification do
+            xml.LabelStockSize do
+              xml.Height(label_size[0])
+              xml.Width(label_size[1])
+            end
+
             xml.LabelPrintMethod do
-              xml.Code('GIF')
+              xml.Code(label_format)
             end
-            xml.HTTPUserAgent('Mozilla/4.5') # hmmm
-            xml.LabelImageFormat('GIF') do
-              xml.Code('GIF')
+
+            xml.HTTPUserAgent('Mozilla/4.5')
```

Comment:
Might be best to put this inside of the `if label_format == 'GIF'` block with a comment about how it's a stupid thing that UPS requires